org.eclipse.scout.rt.rest.jersey.test/src/main/java/org/eclipse/scout/rt/rest/jersey/fixture/ChunkedDataResource.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2010, 2025 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.rest.jersey.fixture;
+
+import static org.junit.Assert.*;
+
+import java.util.Date;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.GenericType;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.scout.rt.dataobject.fixture.FixtureDateId;
+import org.eclipse.scout.rt.dataobject.fixture.FixtureStringId;
+import org.eclipse.scout.rt.dataobject.fixture.FixtureUuId;
+import org.eclipse.scout.rt.platform.BEANS;
+import org.eclipse.scout.rt.platform.job.Jobs;
+import org.eclipse.scout.rt.platform.util.SleepUtil;
+import org.eclipse.scout.rt.rest.IRestResource;
+import org.eclipse.scout.rt.rest.chunked.IChunkedDataWriter;
+import org.glassfish.jersey.server.ChunkedOutput;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("chunked")
+public class ChunkedDataResource implements IRestResource {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ChunkedDataResource.class);
+
+  /**
+   * Implementation example based on JAX-RS API and Scout API
+   *
+   * @see org.eclipse.scout.rt.rest.chunked.IChunkedDataWriter
+   */
+  @GET
+  @Path("string-scout")
+  @Produces(MediaType.TEXT_PLAIN)
+  public GenericType<String> getStringsScout() {
+    @SuppressWarnings("resource")
+    IChunkedDataWriter<String> writer = IChunkedDataWriter.create(String.class, "\n\n");
+    Jobs.schedule(() -> {
+      try {
+        for (int i = 1; i <= 3; i++) {
+          writer.write(Integer.toString(i));
+          LOG.info("Wrote chunk '{}'", i);
+          SleepUtil.sleepSafe(10, TimeUnit.MILLISECONDS);
+        }
+      }
+      finally {
+        writer.close();
+      }
+    }, Jobs.newInput());
```

Comment:
```suggestion
      assertTrue(writer.isClosed);
    }, Jobs.newInput());
```

All methods for this resource seem to be quite similar, actually it would also be possible to generalize these methods (save some lines of code, however probably makes it more difficult to read..), e.g. something like:

```java
  @GET
  @Path("string-scout")
  @Produces(MediaType.TEXT_PLAIN)
  public GenericType<String> getStringsScout() {
    return getScoutInternal(IChunkedDataWriter.create(String.class, "\n\n"), Integer::toString, IChunkedDataWriter::toEntity);
  }

  protected <W extends IChunkedDataWriter<T>, T, R> R getScoutInternal(W writer, IntFunction<T> writeSupplier, Function<W, R> returnFunction) {
    Jobs.schedule(() -> {
      try {
        for (int i = 1; i <= 3; i++) {
          writer.write(writeSupplier.apply(i));
          LOG.info("Wrote chunk '{}'", i);
          SleepUtil.sleepSafe(10, TimeUnit.MILLISECONDS);
          assertFalse(writer.isClosed());
        }
      }
      finally {
        writer.close();
      }
      assertTrue(writer.isClosed());
    }, Jobs.newInput());
    return returnFunction.apply(writer);
  }
```

Reply:
Done. Added writeAsync method, handling writing job. 
Leaved definition of writer and return of response in separate test methods due to readability considerations.